### PR TITLE
Add support for Mattermost Operator deployment via Helm chart

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
- Add support for Mattermost Operator deployment via Helm chart

This PR adds a Helm chart for the Mattermost Operator, with the purpose to make Mattermost deployments faster and easier.

### CODE REVIEW

**Review by gabrieljackson — APPROVED** [review]
Even though this is a PR for a helm chart, it's well done. 😉

File: charts/mattermost-operator/crds/crd-mattermostrestoredbs.yaml
Comment (cpanato) [inline charts/mattermost-operator/crds/crd-mattermostrestoredbs.yaml:1]: can remove this blank line

File: charts/mattermost-operator/templates/mattermost-operator/deployment.yaml
Comment (cpanato) [inline charts/mattermost-operator/templates/mattermost-operator/deployment.yaml:5]: missing labels 
and for all the other manifests I recommend to follow this guide for the labels and annotations
https://helm.sh/docs/chart_best_practices/labels/

**Review by cpanato — CHANGES_REQUESTED** [review]

File: charts/mattermost-operator/templates/mattermost-operator/deployment.yaml
Comment (stylianosrigas) [reply charts/mattermost-operator/templates/mattermost-operator/deployment.yaml:5]: Thanks added labels :)

File: charts/mattermost-operator/templates/mattermost-operator/deployment.yaml
Comment (cpanato) [inline charts/mattermost-operator/templates/mattermost-operator/deployment.yaml:16]: the above labels should be applied here as well

File: charts/mattermost-operator/templates/mattermost-operator/deployment.yaml
Comment (cpanato) [inline charts/mattermost-operator/templates/mattermost-operator/deployment.yaml:20]: and here

Comment (cpanato) [reply charts/mattermost-operator/templates/mattermost-operator/deployment.yaml:20]: see: https://helm.sh/docs/chart_best_practices/pods/#podtemplates-should-declare-selectors

**Review by cpanato — CHANGES_REQUESTED** [review]

File: charts/mattermost-operator/templates/mattermost-operator/deployment.yaml
Comment (stylianosrigas) [reply charts/mattermost-operator/templates/mattermost-operator/deployment.yaml:16]: Thanks my friend, nice catch, done :)

**Comment by stylianosrigas** [comment]
@spirosoik @angeloskyratzakos @Szymongib  Can I get another review please? 💙

File: charts/mattermost-operator/Chart.yaml
Comment (stafot) [inline charts/mattermost-operator/Chart.yaml:3]: vvvv minor: Should change a bit the description

Comment (stylianosrigas) [reply charts/mattermost-operator/Chart.yaml]: Thanks @stafot, fixed :)

File: charts/mattermost-operator/templates/mattermost-operator/clusterrole.yaml
Comment (Szymongib) [inline charts/mattermost-operator/templates/mattermost-operator/clusterrole.yaml:12]: Did you intend to add permissions for `leases` latter, after Operator release?

**Review by Szymongib — APPROVED** [review]
LGTM!

File: charts/mattermost-operator/templates/mattermost-operator/clusterrole.yaml
Comment (stylianosrigas) [reply charts/mattermost-operator/templates/mattermost-operator/clusterrole.yaml:12]: @Szymongib yeap, I was waiting for the release to update also the image and CRDs. All are up to date now :)

File: charts/mattermost-operator/README.md
Comment (angeloskyratzakos) [inline charts/mattermost-operator/README.md:30]:
```suggestion
To start, copy [mattermost-operator/charts/mattermost-operator/values.yaml](https://github.com/mattermost/mattermost-operator/blob/master/charts/mattermost-operator/values.yaml) and name it `config.yaml`. This will be your configuration file for the Mattermost Operator chart. You can use the default values that will deploy Mattermost-Operator, Mysql-Operator and Minio-Operator together (use of mysql and minio operators is not suggested for production environments) or update accordingly.
```

File: charts/mattermost-operator/README.md
Comment (angeloskyratzakos) [inline charts/mattermost-operator/README.md:64]: maybe to put a `helm ls -A` to avoid the namespace section?
i.e.
```suggestion
If you are done with your deployment and want to delete it, use `helm delete <your-release-name>`. If you don't know the name of your release, use `helm ls -A` to find it across all namespaces.
```

Comment (stylianosrigas) [reply charts/mattermost-operator/README.md:64]: We need the namespace to delete the Helm chart.

Comment (stafot) [reply charts/mattermost-operator/README.md:64]: Agree. I think we should add a combination of both syntaxes to be more accurate.

File: charts/mattermost-operator/values.yaml
Comment (stafot) [inline charts/mattermost-operator/values.yaml:142]: do we want these exposed?

**Review by stafot — APPROVED** [review]
Looks very good to me! Nice work, @stylianosrigas

File: charts/mattermost-operator/README.md
Comment (angeloskyratzakos) [reply charts/mattermost-operator/README.md:64]: yeap you are right I have updated my suggestion

File: charts/mattermost-operator/values.yaml
Comment (stylianosrigas) [reply charts/mattermost-operator/values.yaml:142]: Yeap they are the default mySQLoperator values that the user can change.

File: charts/mattermost-operator/README.md
Comment (stylianosrigas) [reply charts/mattermost-operator/README.md:64]: Yeap makes sense now, thanks :)

**Review by angeloskyratzakos — APPROVED** [review]
LGTM, very nice work Stelios! well done!

**Review by cpanato — APPROVED** [review]
looks great!

thanks  for adding this

**Pull request merged**
